core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -461,10 +461,11 @@ private void logFailedEvents(final String topicId, final String eventType, final
                     }
                 }));
             }
-            latch.await(createSendTimeout(), TimeUnit.MILLISECONDS);
-
-            final List<NakadiRecordMetadata> resps =
-                    prepareResponse(nakadiRecords, responses, null);
+            final boolean recordsPublished = latch.await(createSendTimeout(), TimeUnit.MILLISECONDS);
+            final List<NakadiRecordMetadata> resps = prepareResponse(
+                    nakadiRecords, responses,
+                    recordsPublished ? null : new TimeoutException(
+                            "timeout waiting for events to be sent to kafka"));
```

Comment:
Since we are doing it, we could also mention the timeout duration in the message...

Reply:
I did it in the first iteration, but decided to remove it, because of the reduced readability of the code and the same value in the message